src/packages/recompose/isReferentiallyTransparentFunctionComponent.js

```diff
@@ -5,7 +5,7 @@ const isReferentiallyTransparentFunctionComponent = Component => Boolean(
   !isClassComponent(Component) &&
   !Component.defaultProps &&
   !Component.contextTypes &&
-  !Component.propTypes
+  (!Component.propTypes || process.env.NODE_ENV === 'production')
```

Comment:
Yes, I've just tried [UglifyJS](https://skalman.github.io/UglifyJS-online/), both `true || foo` and `foo || true` work.